src/dmd/semantic3.d

```diff
@@ -1295,6 +1295,75 @@ private extern(C++) final class Semantic3Visitor : Visitor
             sc = sc.pop();
         }
 
+        // Check `extern(C++)` functions for invalid the return/parameter types
+        if (funcdecl._linkage == LINK.cpp)
+        {
+            static bool checkCppNonMappableType(Type type, Parameter param = null, Type origType = null)
+            {
+                if (origType is null)
+                {
+                    origType = type;
+                    type = type.toBasetype();
```

Comment:
Now this function is nested in the FuncDeclaration visitor, can be a bit smarter with the use of toBasetype.